flutter_frontend_server/lib/server.dart

```diff
@@ -169,3 +182,82 @@ Future<int> starter(
   frontend.listenAndCompile(compiler, input ?? stdin, options, completer);
   return completer.future;
 }
+
+/// A [RecursiveVisitor] that replaces [Object.toString] overrides with
+/// `super.toString()`.
+class ToStringVisitor extends RecursiveVisitor<void> {
+  /// The [packageUris] must not be null.
+  ToStringVisitor(this._packageUris) : assert(_packageUris != null);
+
+  /// A set of package URIs to apply this transformer to, e.g. 'dart:ui' and
+  /// 'package:flutter/foundation.dart'.
+  final Set<String> _packageUris;
+
+  /// Turn 'dart:ui' into 'dart:ui', or
+  /// 'package:flutter/src/semantics_event.dart' into 'package:flutter'.
+  String _importUriToPackage(Uri importUri) => '${importUri.scheme}:${importUri.pathSegments.first}';
+
+  bool _isInTargetPackage(Procedure node) {
+    return _packageUris.contains(_importUriToPackage(node.enclosingLibrary.importUri));
+  }
+
+  bool _hasKeepAnnotation(Procedure node) {
+    for (ConstantExpression expression in node.annotations.whereType<ConstantExpression>()) {
+      final InstanceConstant constant = expression.constant as InstanceConstant;
+      if (constant.classNode.name == 'pragma' && constant.classNode.enclosingLibrary.importUri.toString() == 'dart:core') {
+        final StringConstant pragmaName = constant.fieldValues.values.first as StringConstant;
```

Comment:
`constant.fieldValues.values.first` doesn't look reliable as there are multiple fields which may go in an arbitrary order in constant.fieldValues map.

Reply:
Aww phooey.  I'll try to fix that up.  My hope was that since this should be a stable order map (the contract for Map is that order remains the same), and we know the type has its first parameter as the name, this could be ok.  But I'll try to do something better here.

Reply:
Done.